File: backend/pydantic_models/pill_user.py
from pydantic import BaseModel
from datetime import date
from typing import Optional


class PillUserResponse(BaseModel):
    id_pill_user: int
    name: str
    active_substance: Optional[str]
    expiration_date: Optional[date]
    category: Optional[str]
    intake_type: Optional[str]
    quantity: Optional[float]
    format: Optional[str]
    dosage: Optional[str]
    comments: Optional[str]
    image_url: Optional[str]
    last_price: Optional[float]

    class Config:
        from_attributes = True

class PillUserCreate(BaseModel):
    name: str
    active_substance: Optional[str] = None
    expiration_date: Optional[date] = None
    category: Optional[str] = None
    intake_type: Optional[str] = None
    quantity: Optional[float] = None
    format: Optional[str] = None
    dosage: Optional[str] = None
    comments: Optional[str] = None
    image_url: Optional[str] = None
    last_price: Optional[float] = None

    class Config:
        from_attributes = True

File: frontend/lib/models/pill_user.dart
import 'dart:convert';
import 'dart:developer';
import 'package:http/http.dart' as http;

class PillUser {
  final int idPillUser;
  String name;
  DateTime? expirationDate;
  String? activeSubstance;
  String? category;
  String? intakeType;
  double? quantity;
  String? format;
  String? dosage;
  String? comments;
  String? imageUrl;
  double? lastPrice;

  PillUser({
    required this.idPillUser,
    required this.name,
    this.expirationDate,
    this.activeSubstance,
    this.category,
    this.intakeType,
    this.quantity,
    this.format,
    this.dosage,
    this.comments,
    this.imageUrl,
    this.lastPrice,
  });

  Map<String, dynamic> toJson() {
    return {
      'id_pill_user': idPillUser,
      'name': name,
      'expiration_date': expirationDate?.toIso8601String(),
      'active_substance': activeSubstance,
      'category': category,
      'intake_type': intakeType,
      'quantity': quantity,
      'format': format,
      'dosage': dosage,
      'comments': comments,
      'image_url': imageUrl,
      'last_price': lastPrice,
    };
  }

  factory PillUser.fromJson(Map<String, dynamic> json) {
    return PillUser(
      idPillUser: json['id_pill_user'],
      name: json['name'],
      activeSubstance: json['active_substance'],
      expirationDate: json['expiration_date'] != null
          ? DateTime.parse(json['expiration_date'])
          : null,
      category: json['category'],
      intakeType: json['intake_type'],
      quantity: json['quantity'],
      format: json['format'],
      dosage: json['dosage'],
      comments: json['comments'],
      imageUrl: json['image_url'],
      lastPrice: json['last_price'],
    );
  }

  @override
  String toString() {
    return 'PillUser{idPillUser: $idPillUser, name: $name, expirationDate: $expirationDate, quantity: $quantity}';
  }
}